pkg/trigger/aws/handler.go

```diff
@@ -29,17 +30,25 @@ func (h *HookHandler) Loop(nodeName string) {
 			continue
 		}
 		glog.Infof("Status of instance '%v' is '%v', autoscaling group is '%v'", h.AutoScaling.Options.InstanceID, *status, *autoScalingGroupName)
-		if !h.AutoScaling.IsTerminating(status) {
+		if !h.AutoScaling.IsTerminating(status) && !h.AutoScaling.IsTerminatingWait(status) {
 			continue
 		}
 
-		err = h.Drainer.Drain(nodeName)
-		if err != nil {
-			glog.Warningf("Not all pods on this host can be evicted, will try again: %s", err)
+		if !drained {
```

Comment:
so if I understand correctly, there was a misleading message in logs, that was not always true, and now we fix that, right?

Reply:
Well, after moving the draining to happen as soon as it sees the "Termination" state, it will probably loop several times before it hits the "Termination:Wait" state, and this is to just avoid it trying to cordon and drain again if it's already done so successfully.